src/librawspeed/common/Memory.h

```diff
@@ -69,6 +69,17 @@ alignedCalloc(size_t nmemb, size_t size) {
   return alignedMalloc<alignment>(size);
 }
 
+template <size_t alignment, typename T>
+// coverity[+alloc]
+inline void* __attribute__((malloc, warn_unused_result))
+alignedCalloc(size_t nmemb) {
```

Comment:
Please note that the term `calloc` is used in the standard with the explicit semantic meaning of zero initializing the buffer (http://en.cppreference.com/w/cpp/memory/c/calloc) which this code does not. And I think it should not. Or maybe it should but then this function should not be used where this initialization is not required and potentially a waste of cpu cycles. Maybe just call it `alignedArrayAlloc` or something like that?

Also, if your function already knows the type of the objects, I'd return a pointer to that type instead of void. Should make a bunch of c-casts unnecessary.

I've been looking into some sse optimizations of the jpeg decoding. It has not been as beneficial as I hoped, so it is sitting on my desk for now. But while doing that I prepared an aligned allocator class to be able to use a std::vector with aligned memory. I have not checked, but I guess most of the current usages of this 'manual' alloc/free code could be improved by replacing it with a std::vector buffer?